### PR TITLE
Week2 Homework - Oguzhan

### PR DESCRIPTION
**Pull request by oguzhanturgut**
- maartjes-work.js
computeEarnings function with filter and reduce chaining
- map-filter.js
doubleOddNumbers function with filter map chaining or reduce

### CODE REVIEW

File: Week2/homework/map-filter.js [hunk 1]
```diff
@@ -1,8 +1,16 @@
 'use strict';
 
 function doubleOddNumbers(numbers) {
-  // Replace this comment and the next line with your code
-  console.log(numbers);
+  // the best practice
+  return numbers
+    .filter(number => Number.isInteger(number) && number % 2 !== 0)
```
Comment (yash-kapila) [inline Week2/homework/map-filter.js]: 👍

File: Week2/homework/maartjes-work.js [hunk 2]
```diff
@@ -44,18 +44,29 @@ const tuesday = [
 
 const maartjesTasks = monday.concat(tuesday);
 const maartjesHourlyRate = 20;
+const timeThreshold = 120;
+
+// function computeEarnings(tasks, hourlyRate) {
+//   return tasks
```
Comment (yash-kapila) [inline Week2/homework/maartjes-work.js]: I see what you are trying to do here and it works fine. However, you missed one small point from homework which is `Map the tasks to durations in hours.`. It's equally important that you follow all instructions provided in the homework. 😉

File: Week2/homework/maartjes-work.js [hunk 3]
```diff
@@ -44,18 +44,29 @@ const tuesday = [
...
+//     .filter(task => task.duration >= timeThreshold)
+//     .reduce((acc, current) => acc + (current.duration / 60) * hourlyRate, 0);
```
Comment (yash-kapila) [inline Week2/homework/maartjes-work.js]: Wouldn't `task` be a better variable name than `current` here?

Comment (oguzhanturgut) [reply Week2/homework/maartjes-work.js]: Definitely yes. Changed to `task`

File: Week2/homework/maartjes-work.js (hunk 2 above)
Comment (oguzhanturgut) [reply Week2/homework/maartjes-work.js]: I missed that point and shifted that step in `reduce` part. Now added `map` step for that requirement. Thanks for drawing attention to that point

**Comment by oguzhanturgut** [comment]
> Hi @oguzhanturgut - I've finished reviewing your homework and am approving the same. See you Sunday.

Hi @yash-kapila . Thanks for your time reviewing the homework and valuable feedbacks. See you on Sunday🖐

**Pull request closed without merging**